pkg/gardenlet/controller/shoot/care/health.go

```diff
@@ -666,6 +679,32 @@ func (h *Health) CheckClusterNodes(
 	return nil, nil
 }
 
+// CheckNodeAgentLeases checks if all given nodes have a corresponding lease
+func CheckNodeAgentLeases(nodeList *corev1.NodeList, leaseList *coordinationv1.LeaseList, clock clock.Clock) error {
+	if len(leaseList.Items) == 0 {
+		return fmt.Errorf("no leases")
+	}
+
+	leases := map[string]coordinationv1.Lease{}
+	for _, lease := range leaseList.Items {
+		nodeName := strings.ReplaceAll(lease.Name, gardenerutils.NodeLeasePrefix, "")
+		leases[nodeName] = lease
```

Comment:
This `leaseList` could also contain `Lease`s not maintained by `gardener-node-agent`, so let's ignore those.

```suggestion
		if strings.HasPrefix(lease.Name, gardenerutils.NodeLeasePrefix) {
			nodeName := strings.ReplaceAll(lease.Name, gardenerutils.NodeLeasePrefix, "")
			nodeNameToLease[nodeName] = lease
		}
```